pkg/metrics/tags_test.go

```diff
@@ -224,6 +227,35 @@ func TestContexts(t *testing.T) {
 	}
 }
 
+func BenchmarkPodRevisionContext(b *testing.B) {
+	for _, cache := range []bool{true, false} {
+		b.Run(fmt.Sprintf("sequential-cache-%v", cache), func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				rev := "name"
+				if !cache {
+					rev = rev + strconv.Itoa(i)
+				}
+
+				PodRevisionContext("pod", "container", "ns", "svc", "cfg", rev)
+			}
+		})
+
```

Comment:
Not sure this tests what we need. One just creates N contexts the other creates 1. What it does not take into account is cache work needed to take care of N revisions, including evictions etc.
Perhaps a better one is to:
```
if cache {
   Make the same call again
}
```
and compare the additional cost of making a second call vs when all the calls are for a new revision?

Reply:
hm.. I mean it tests the two cases that actually happen right? service in cache / not in cache? I mean you're right that in the cache case the first request _technically_ isn't in cache, but the next 2 million-sh are :) I can add a line to explicitly add to cache first to clear that up? The never hitting cache case covers evictions because it runs way more than 4096 times..

Reply:
We did some conversations in slack :)
Not sure what we agreed on, but along the lines that we might want to expand this one, but may be later.
The main issue that I have with this particular implementation is that it presumes cache performance is static across the whole range of its size and with random lookups (given that it needs to rearrange nodes for LRU behavior etc).
I.e. we're comparing worst noncache performance (actually worse since we actually cache things but never use the cached value) with best possible cached performance — giving us a a rosier than real life view

😊

Reply:
okie, new version does always-hit, hit-but-at-capacity, and way-over-capacity-of-lru